lib/job.js

```diff
@@ -204,21 +204,26 @@ Job.prototype.promote = function(){
     });
 };
 
+/**
+ * Attempts to retry the job. Only a job that has failed can be retried.
+ *
+ * @return {Promise} If resolved and return code is 1, then the queue emits a waiting event
+ * otherwise the operation was not a success and throw the corresponding error. If the promise
+ * rejects, it indicates that the script failed to execute
+ */
 Job.prototype.retry = function(){
```

Comment:
One consideration on that though. The current implementation works with failed jobs only, so we'll be breaking some expectations on `Job#retry` by letting it to retry completed job as well.

One common usage scenario I think is having parallel workers that have some kind of retry logic for failed jobs, something like this:

```javascript
setInterval(() => {
  queue.getFailed().then(jobs => {
    return Promise.each(jobs, job => job.retry());
  });
}, 60 * 1000);
```

The above may lead to problems because if another worker retried a job successfully, it might be retried yet again by another worker since that job is also retriable if it's completed.

Perhaps we can allow parameters to `retry` to drive the behavior? or expose `repeat` as a separate Job function instead?